Adding an override to LoggerProvider.GetLogger

juliuskoval opened the pull request:
Towards https://github.com/open-telemetry/opentelemetry-dotnet/issues/4433
`LoggerProvider.GetLogger` should accept attributes according to the [spec.](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#logs)

## Changes

I added an overload to the `LoggerProvider.GetLogger` method which accepts attributes and I modified the `ProtobufOtlpLogSerializer` so that it now exports both the version of instrumentation scope and the associated attributes.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

CodeBlanch commented on src/OpenTelemetry.Api/Logs/Logger.cs:
```diff
@@ -63,8 +68,10 @@ public abstract void EmitLog(
+        IEnumerable<KeyValuePair<string, object?>>? attributes)
     {
         this.Version = version;
+        this.Attributes = attributes;
```
If we are going to do this we need to take a copy here so they can't be mutated after being set.

Same idea as here, which also sorts:

https://github.com/dotnet/runtime/blob/fcd9339f16500a4954d84e9d0d93a99e84b08987/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L42-L49

juliuskoval commented:
I added a benchmark since @Kielek asked me to [here](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4433#issuecomment-2658702231). Sorting the instrumentation scope attributes increases the performance impact in the benchmark I added by about 40%, so I'm not sure if it's worth it.
